spec/std/indexable_spec.cr

```diff
@@ -1,6 +1,35 @@
 require "spec"
 require "spec/helpers/iterate"
 
+module SomeInterface; end
+
+private record One do
+  include SomeInterface
+end
+
+private record Two do
+  include SomeInterface
+end
+
+private struct InterfaceIndexable(T, S)
+  include Indexable(T)
+
+  @values = [One.new, Two.new]
+
+  def size : Int32
+    S
+  end
+
+  def unsafe_fetch(index : Int) : T
+    case index
+    when 0 then @values[0]
+    when 1 then @values[1]
+    else
+      raise ""
+    end
+  end
+end
```

Comment:
**suggestion:** This helper type could be simplified quite a bit. There's no need for generic arguments or instance variables.
```suggestion
private struct InterfaceIndexable
  include Indexable(SomeInterface)
  
  def size
    2
  end

  def unsafe_fetch(index : Int) : SomeInterface
    case index
    when 0 then One.new
    when 1 then Two.new
    else
      raise ""
    end
  end
end
```